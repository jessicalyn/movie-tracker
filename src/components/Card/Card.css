.card {
    display: flex;
    flex-direction: column;
    position:relative;
    width:20%;
    margin-top:20px;
    margin-right:2px;
    height:500px;
}
.card button {
    width: 50px;
    height: 50px;
    border-radius: 50px;
    position: absolute;
    top: 2%;
    right: 5%;
    background: none;
    border: none;
    cursor: pointer;
} 
.card button:focus { 
    outline: none; 
}
.card img {
    width:100%;
    height: 500px;
}
.card button img{
    width:40px;
    height:40px;
}

.card h3{
    color:#1f1f1f;
}
.card p{
    color: #ffff;
}