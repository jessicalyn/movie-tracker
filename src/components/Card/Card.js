import React from "react";
import { Component } from "react";
import { connect } from "react-redux";
import './Card.css'
import heart from '../../images/add.png'

export class Card extends Component {
  constructor(props) {
    super(props);
    this.state = {
      message: ""
    }
  }

  addFavorites = async () => {
    if (this.props.user.id) {
      return await this.fetchFavorites();
    } else {
      this.setState({
        message: "Please login or sign up to favorite movies."
      })
    }
  }
  
  fetchFavorites = async () => {
    const { movie, user } = this.props;
    let data = {
      movie_id: movie.id,
      user_id: user.id,
      title: movie.title,
      poster_path: movie.poster_path,
      release_date: movie.release_date,
      vote_average: movie.vote_average,
      overview: movie.overview,
    }
    
    const url = "http://localhost:3000/api/users/favorites/new"
    try{
    const addFav = await fetch(url, {
      method: "POST",
      body: JSON.stringify(data),
      headers: {
        "Content-Type": "application/json"
      }
    })
    
    const response = await addFav.json();
    this.setState({
      message: response.message
    })
  }  catch(error){
      this.setState({
        message: "Sorry something went wrong, please try again."
      })
    }
  }

  render() {
    const { movie } = this.props
    const poster = movie.poster_path
    const path = `https://image.tmdb.org/t/p/w185/${poster}`
    return (
      <section className="card">
        {/* <h3>{movie.title}</h3> */}
        <img src={path} alt={movie.title} />
        <button onClick={this.addFavorites}>
          <img src={heart}></img>
        </button>
        <p>{this.state.message && this.state.message}</p>
      </section>
    )
  }
}

const mapStateToProps = state => ({
    user: state.user
})

export default connect(mapStateToProps)(Card)
