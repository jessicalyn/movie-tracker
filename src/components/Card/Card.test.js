import React from 'react';
import ReactDOM from 'react-dom';
import { Card, mapStateToProps } from './Card'
import { shallow } from 'enzyme';

describe('Card', ()=> {
    describe('Login', () => {
        let wrapper; 
        beforeEach(() => {
            wrapper = shallow(<Card/>)
        })

        it('should match the snapshot with all data passed in', () => {
            expect(wrapper).toMatchSnapshot()
        })
    })
})