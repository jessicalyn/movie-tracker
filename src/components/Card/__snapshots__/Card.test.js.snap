// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Login should match the snapshot with all data passed in 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    movie={
      Object {
        "movie_id": 1,
        "overview": "movie.overview",
        "poster_path": "movie.poster_path",
        "release_date": 0.00033085194375516956,
        "title": "movie.title",
        "vote_average": "movie.vote_average",
      }
    }
    user={
      Object {
        "id": 1,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3>
          movie.title
        </h3>,
        <img
          alt="movie.title"
          src="https://image.tmdb.org/t/p/w185/movie.poster_path"
        />,
        <button
          onClick={[Function]}
        >
          Favorite
        </button>,
        "",
      ],
      "className": "card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "movie.title",
        },
        "ref": null,
        "rendered": "movie.title",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "movie.title",
          "src": "https://image.tmdb.org/t/p/w185/movie.poster_path",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Favorite",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Favorite",
        "type": "button",
      },
      "",
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3>
            movie.title
          </h3>,
          <img
            alt="movie.title"
            src="https://image.tmdb.org/t/p/w185/movie.poster_path"
          />,
          <button
            onClick={[Function]}
          >
            Favorite
          </button>,
          "",
        ],
        "className": "card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "movie.title",
          },
          "ref": null,
          "rendered": "movie.title",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "movie.title",
            "src": "https://image.tmdb.org/t/p/w185/movie.poster_path",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Favorite",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Favorite",
          "type": "button",
        },
        "",
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
